lokimq/bt_serialize.h

```diff
@@ -909,5 +922,192 @@ class bt_dict_consumer : private bt_list_consumer {
     bt_dict_consumer consume_dict_consumer() { return consume_dict_data(); }
 };
 
+/// Currently this is not idiot-proof.  That is to say, the compiler won't catch everything and
+/// you technically can create invalid BEncoded strings.  Basically you can add list elements
+/// to a dict and dict elements to a list, and obviously that can end badly.  This may be fixed
+/// at a later time.
```

Comment:
These /// comments are documentation; they really should describe the type in general before it describes idiots.

Reply:
Yeah, fair cop.  WIP indeed...